metadata.rb

```diff
@@ -3,12 +3,16 @@
 maintainer_email 'cookbooks@chef.io'
 license 'Apache-2.0'
 description 'Installs the chef-vault gem and provides chef_vault_item recipe helper'
-version '3.1.1'
+version '3.1.2'
 
 source_url 'https://github.com/chef-cookbooks/chef-vault'
 issues_url 'https://github.com/chef-cookbooks/chef-vault/issues'
 chef_version '>= 12.9'
 
-gem 'chef-vault'
+if `chef-client --version`.split[1].to_f < 13
```

Comment:
Shelling out here seems incredibly problematic. Also, I think what you really want to do it skip the gem install entirely on modern chef releases where we just shipped the gem built into the client.

Reply:
What you probably want here is

Gem::Requirement.new('< 13').satisfied_by?(Gem::Version.new(Chef::VERSION))

Reply:
Can you explain what this code does? I'm assuming this replaces the entire if statement?

My knowledge of ruby isn't that extensive

Reply:
This code `Gem::Requirement.new('< 13').satisfied_by?(Gem::Version.new(Chef::VERSION))` it will get the current installed chef-client version and validate it on the condition, so:
```rb
if Gem::Requirement.new('< 13').satisfied_by?(Gem::Version.new(Chef::VERSION))
# code
end
```
Will get the current installed chef version and if it is < 13 so 12 and less will go to the `IF` and apply your code.
It will compare the requirements you specified `<13` against the installed chef version `Chef::VERSION`. it is better than what you already has now by running `chef-client --version` this is more closer to the chef code style :)